keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/AdminService.kt

```diff
@@ -113,4 +116,37 @@ class AdminService(
     }
   }
 
+  fun forceSkipArtifactVersion(application: String, environment: String, artifactReference: String, version: String) {
+    val deliveryConfig = repository.getDeliveryConfigForApplication(application)
+    val artifact = deliveryConfig.matchingArtifactByReference(artifactReference)
+      ?: throw UserException("application $application contains no artifact ref $artifactReference. Artifact references are: ${deliveryConfig.artifacts.map { it.reference }}")
+
+     // Identify the current version in the environment
+    val currentVersion = repository.getCurrentArtifactVersions(deliveryConfig, environment)
+      .firstOrNull { it.reference == artifactReference }
+      ?: throw UserException("application $application has no artifact with reference $artifactReference in environment $environment in CURRENT status")
```

Comment:
Can we ever be in a state where there is no `current` artifact but one is stuck in deploying?

Reply:
I think it is possible for the system to be in that state (e.g., if the first deploy gets stuck). If that happens, I'm not sure what state we would want to force the stuck version into. "Skipped" doesn't feel like the right state.

I think what I'll do is also add an API endpoint that lets the admin choose what state to force the version into. That way, the operator can decide in the moment what makes the most sense.

Reply:
Okay - but this api call would fail because a precondition is not true (there is no current version). We can either deal with that later, or you could add a `force: Boolean` flag which does not care if there is no current version